File: ios/CarMechanicAR/Models/CarPart.swift
import Foundation

struct CarPart: Identifiable, Hashable, Codable {
    let id: String
    let nameFa: String
    let nameEn: String
    let category: String
    let shortDescription: String
    let fullDescription: String
    let function: String
    let location: String
    let commonSymptoms: [String]
    let maintenanceTips: [String]
    let difficultyLevel: String
    let safetyWarning: String?
    let imageName: String
    let model3DName: String
    let videoURL: URL?
    let relatedParts: [String]
}

struct DetectedCarPart: Identifiable, Hashable {
    let id = UUID()
    let partID: String
    /// Vision normalized coordinate system, origin at lower-left.
    let boundingBox: CGRect
    let confidence: Float
    let timestamp: Date
}

File: android/app/src/main/java/com/azadi/carmechanicar/model/CarPart.kt
package com.azadi.carmechanicar.model

data class CarPart(
    val id: String,
    val nameFa: String,
    val nameEn: String,
    val category: String,
    val shortDescription: String,
    val function: String,
    val location: String,
    val safetyWarning: String? = null
)

data class DetectedCarPart(
    val partId: String,
    val left: Float,
    val top: Float,
    val right: Float,
    val bottom: Float,
    val confidence: Float,
    val timestamp: Long = System.currentTimeMillis()
)
